src/pysweepme/Ports.py

```diff
@@ -959,58 +971,95 @@ def open_internal(self):
         else:
             self.read_termination = ""
 
-        self.last_writetime = time.time()
+        self.last_write_time = time.time()
 
-    def close_internal(self):
+    def close_internal(self) -> None:
+        """Closing the socket."""
         self.port.close()
 
-    def get_identification(self):
-
+    def get_identification(self) -> str:
+        """Returns the *IDN of the socket."""
         self.write("*IDN?")
         return self.read()
 
-    def write_internal(self, cmd: str):
+    def clear_internal(self) -> None:
+        """Read out the buffer until its empty."""
+        # Set a 1ms timeout for maximum read out speed, but still >0 to avoid non-blocking (which might not read out all data)
+        timeout = self.port.gettimeout()
+        self.port.settimeout(0.001)
+
+        with contextlib.suppress(socket.timeout):
+            while True:
+                data = self.read_chunk(4096)
+                if not data:
+                    break
+
+        # Set the timeout back to the original value
+        self.port.settimeout(timeout)
+        self.clear_buffer()
```

Comment:
If there is an exception other than socket.timeout, the function will not reset the timeout to it's previous value. Maybe use a try-finally here, like this (not tested):
```suggestion
        timeout = self.port.gettimeout()
        try:
            self.port.settimeout(0.001)
            with contextlib.suppress(socket.timeout):
                while True:
                    data = self.read_chunk(4096)
                    if not data:
                        break
            self.clear_buffer()
        finally:
            # Set the timeout back to the original value
            self.port.settimeout(timeout)
```